### PR TITLE
chore(dal): Ensure that all different get_code Language types are covered

### PR DESCRIPTION
**Pull request by stack72**
This introduces a new test schema - Katy Perry - it's all codegen funcs and no actions :D

### CODE REVIEW

**Comment by stack72** [comment]
bors try

File: lib/dal-test/src/schemas/test_exclusive_schema_katy_perry.rs [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+            }";
+    let yaml_code_gen_func =
+        build_codegen_func(yaml_codegen_func_code, yaml_codegen_fn_name).await?;
+
+    // Build string CodeGen Func
+    let string_codegen_fn_name = "test:generateStringCode";
+    let string_codegen_func_code = "async function main(input: Input): Promise < Output > {
+                return {
+                    format: \"string\",
+                    code: \"poop canoe\"
```
Comment (nickgerace) [inline lib/dal-test/src/schemas/test_exclusive_schema_katy_perry.rs:47]: Yeah, we're never gonna lose this

File: lib/dal-test/src/schemas/test_exclusive_schema_katy_perry.rs [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+
+    // Author Resource Payload Func
+    let resource_payload_to_value_func = build_resource_payload_to_value_func().await?;
+
+    // Build YAML CodeGen Func
+    let yaml_codegen_fn_name = "test:generateYamlCode";
+    let yaml_codegen_func_code = "async function main(input: Input): Promise < Output > {
+                return {
+                    format: \"yaml\",
+                    code: Object.keys(input.domain).length > 0 ? YAML.stringify(input.domain) : \"\"
```
Comment (nickgerace) [inline lib/dal-test/src/schemas/test_exclusive_schema_katy_perry.rs:36]: Good idea!

File: lib/dal-test/src/schemas/test_exclusive_schema_katy_perry.rs [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+
+    let swifty_spec = kp_builder
+        .func(identity_func_spec)
+        .func(resource_payload_to_value_func)
+        .func(yaml_code_gen_func)
+        .func(string_code_gen_func)
+        .schema(kp_schema)
+        .build()?;
+
+    let swifty_pkg = SiPkg::load_from_spec(swifty_spec)?;
```
Comment (nickgerace) [inline lib/dal-test/src/schemas/test_exclusive_schema_katy_perry.rs:116]: Is she Taylor or Katy?? :exploding_head:

File: lib/dal/tests/integration_test/component/get_code.rs [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+
+    let (codegen_view, has_code) = Component::list_code_generated(ctx, component.id())
+        .await
+        .expect("unable to get codegen views");
+
+    assert_eq!(codegen_view.len(), 1);
+    assert!(has_code, "true");
+
+    // This is safe as we would have failed the above test otherwise
+    let codegen = codegen_view.clone().pop().unwrap();
```
Comment (nickgerace) [inline lib/dal/tests/integration_test/component/get_code.rs:26]: Fair enough!

File: lib/dal/tests/integration_test/component/get_code.rs [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+    assert_eq!(yaml_codegen.language, CodeLanguage::Yaml);
+    assert_eq!(yaml_codegen.message, None);
+    assert_eq!(
+        yaml_codegen.code,
+        Some("name: all codegen and no actions\n".to_string())
+    );
+}
+
+#[test]
+async fn get_code_no_codegen_funcs(ctx: &mut DalContext) {
```
Comment (nickgerace) [inline lib/dal/tests/integration_test/component/get_code.rs:79]: Really good regression test idea

File: lib/dal-test/src/schemas/test_exclusive_schema_katy_perry.rs (hunk 3 above)
Comment (stack72) [reply lib/dal-test/src/schemas/test_exclusive_schema_katy_perry.rs:116]: d'oh!! good catch on the naming - fixing now :)

**Comment by stack72** [comment]
bors try

**Comment by stack72** [comment]
bors cancel

**Comment by stack72** [comment]
bors try

**Pull request closed without merging**